pkg/pool-utils/contracts/BasePool.sol

```diff
@@ -273,6 +336,18 @@ abstract contract BasePool is IBasePool, BasePoolAuthorization, BalancerPoolToke
         }
     }
 
+    /**
+     * @notice Remove liquidity from a pool.
+     * @dev This function can only be called from the Vault, from `exitPool`. If this is a special recovery mode exit,
+     * ensure we are in Recovery Mode, and invoke `_recoveryModeExit`. This performs a proportional exit in the simplest
+     * possible manner (e.g., without scaling, so lower-decimal tokens will have less precision). It should not perform
+     * any operations that might fail: no external calls, no iterative calculations, etc.
+     *
+     * This contract implements a default version `_recoveryModeExit`; pools can override it if there are special
+     * considerations, or other operations that need to be performed to maintain pool functions.
+     *
+     * Otherwise, call `_onExitPool`, which each pool type implements to handle all the exits it supports.
```

Comment:
Tbh this feels a bit too involved - a detailed description of internal implementation details. If anything I'd place it inside the code and not in the user facing documentation - but we already have those docs.

Reply:
It's greatly simplified in the current version.